components/cam/src/physics/cam/shoc.F90

```diff
@@ -3129,7 +3061,183 @@ real(rtype) function esatw_shoc(t)
    end if
 end
 
-end module shoc
+
+subroutine compute_brunt_shoc_length(nlev, shcol, dz_zt, thv, thv_zi, brunt) 
+
+  !=========================================================
+  !
+  ! Computes the brunt_visala frequency 
+
+  implicit none
+  integer, intent(in) :: nlev, shcol  
+  ! Grid difference centereted on thermo grid [m]
+  real(rtype), intent(in) :: dz_zt(shcol,nlev)
+  ! virtual potential temperature [K]
+  real(rtype), intent(in) :: thv(shcol,nlev)
+  ! virtual potential temperature [K] at interface 
+  real(rtype), intent(in) :: thv_zi(shcol,nlev) 
+  ! brunt vaisala frequency [s-1]
+  real(rtype), intent(out) :: brunt(shcol, nlev)
+  integer k, i
+
+  do k=1,nlev
+    do i=1,shcol
+      brunt(i,k) = (ggr/thv(i,k)) * (thv_zi(i,k) - thv_zi(i,k+1))/dz_zt(i,k)
+    enddo
+  enddo
+
+end subroutine compute_brunt_shoc_length
+
+subroutine compute_l_inf_shoc_length(nlev,shcol,zt_grid,dz_zt,tke,l_inf)
```

Comment:
Possibly silly question: is this really the L_inf norm? Isn't that the (absolute) max norm for a function integrated over a given interval? I'm likely just missing something.